src/components/Form/ComponentMap.js

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { TextInput } from '../index'
+
+export default function GenerateComponent(componentName, props) {
+  switch (componentName) {
+    case 'TextInput':
+      return <TextInput {...props} />
+    default:
+      throw new Error('whoopsie daisy')
```

Comment:
So this confirms to my mind that your intent is you can only ask for components that exist. So my assertion above about having to maintain the list seems accurate.